### PR TITLE
maven plugin deploys test catalog with or without a galasa.token

### PR DESCRIPTION
**Pull request by techcobweb**
- empty to force a build to run
- galasa-token is optional. It gets used if it is supplied

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: galasa-maven-plugin/src/main/java/dev/galasa/maven/plugin/DeployTestCatalog.java [hunk 1]
```diff
@@ -48,21 +48,26 @@ public class DeployTestCatalog extends AbstractMojo {
...
     public boolean      skipDeployTestCatalog;
 
     // A protected variable so we can inject a mock factory if needed during unit testing.
     protected AuthenticationServiceFactory authFactory = new AuthenticationServiceFactoryImpl();
 
     protected BootstrapLoader bootstrapLoader = new BootstrapLoaderImpl();
 
+    public static final boolean DEFAULT_ASSUMPTION_AUTHENTICATION_ENABLED = false ;
+
+    public static final String BOOTSTRAP_PROPERTY_NAME_IS_AUTH_ENABLED = "framework.auth.isEnabled";
```
Comment (KirbyKatcher) [inline galasa-maven-plugin/src/main/java/dev/galasa/maven/plugin/DeployTestCatalog.java:68]: this constant is no longer required

File: galasa-maven-plugin/src/main/java/dev/galasa/maven/plugin/DeployTestCatalog.java [hunk 2]
```diff
@@ -100,21 +105,36 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 
         URL testcatalogUrl = calculateTestCatalogUrl(bootstrapProperties, this.testStream, this.bootstrapUrl);
 
-        checkGalasaAccessTokenIsValid(this.galasaAccessToken); 
-
-        String jwt = getAuthenticatedJwt(this.authFactory, this.galasaAccessToken, this.bootstrapUrl) ;
+        String jwt = null ;
+        // For now, if no galasa token is supplied, that's ok. It's optional.   
+        if (this.galasaAccessToken==null || this.galasaAccessToken.isEmpty()) {
```
Comment (KirbyKatcher) [inline galasa-maven-plugin/src/main/java/dev/galasa/maven/plugin/DeployTestCatalog.java:110]: The if statement could be condensed to just make sure that if !=null and it not empty then call the getAuthenticatedJwt

File: galasa-maven-plugin/src/main/java/dev/galasa/maven/plugin/DeployTestCatalog.java [hunk 3]
```diff
@@ -100,21 +105,36 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
+            if (DEFAULT_ASSUMPTION_AUTHENTICATION_ENABLED) {
+                getLog().info("Bootstrap properties from server do not include "+BOOTSTRAP_PROPERTY_NAME_IS_AUTH_ENABLED+" so we assume server is insisting on authentication.");
+            } else {
+                getLog().info("Bootstrap properties from server do not include "+BOOTSTRAP_PROPERTY_NAME_IS_AUTH_ENABLED+" so we assume server is not insisting on authentication.");
+            }
+        } else {
+            if (isEnabledPropStr.equalsIgnoreCase("true") ) {
+                getLog().info("Bootstrap properties from server include the property "+BOOTSTRAP_PROPERTY_NAME_IS_AUTH_ENABLED+" so we know the server requires authentication.");
+                isAuthEnabled = true;
+            }
```
Comment (KirbyKatcher) [inline galasa-maven-plugin/src/main/java/dev/galasa/maven/plugin/DeployTestCatalog.java:134]: based on our conversation I think this is no longer needed

File: galasa-maven-plugin/src/test/java/dev/galasa/maven/plugin/DeployTestCatalogTest.java
Comment (KirbyKatcher) [inline galasa-maven-plugin/src/test/java/dev/galasa/maven/plugin/DeployTestCatalogTest.java:1]: I think the tests in here should test if when a token is not supplied it makes a connection without the authorization header and with it if the token is present

**Comment by galasa-team** [comment]
Build successful

**Review by KirbyKatcher — APPROVED** [review]

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**